src/annotator/components/ProfileModal.tsx

```diff
@@ -12,26 +13,37 @@ type ProfileModalProps = {
 };
 
 export function ProfileModal({ eventBus, config }: ProfileModalProps) {
+  const store = useSidebarStore();
   const [isHidden, setIsHidden] = useState(true);
   const emitterRef = useRef<Emitter | null>(null);
+  // Used only to track when was this modal first open, delaying the iframe to
+  // be loaded until strictly necessary.
+  const [isOpen, setIsOpen] = useState(false);
```

Comment:
This probably needs a better naming. Maybe `isOpenedAtLeastOnce`?

Also, I feel `isHidden` and `isOpen` overlap a bit. Would it make sense to completely "unmount" the modal when it is closed so that the iframe (and its content) are loaded again the next time it is opened?

Reply:
On the naming of the first state reference...perhaps `hasOpened`?

We've been slowly moving toward a convention, with naming boolean state references, of omitting `is`, so you might do OK with:

* `[hasOpened, setHasOpened]` and
* `[hidden, setHidden]`

Does that help?

Reply:
It does :)

What about my second question? Would it make sense to use `isHidden` (or whatever we call it) only? And instead of using it to hide/show the component via CSS, use it to return null or return the children, including the iframe.

That would have a side "benefit", which is that the iframe would have fresh content every time it is opened. I've seen we "hack" that in the notebook by refreshing the `key` on every load.